pkg/processor/build/runtime/dotnetcore/runtime.go

```diff
@@ -56,6 +58,10 @@ func (d *dotnetcore) GetProcessorDockerfileInfo(versionInfo *version.Info,
 
 	// set the default base image
 	processorDockerfileInfo.BaseImage = "mcr.microsoft.com/dotnet/core/runtime:3.1"
+	if baseImageRegistry != "" {
+		processorDockerfileInfo.BaseImage =
+			strings.Replace(processorDockerfileInfo.BaseImage, "mcr.microsoft.com", baseImageRegistry, 1)
```

Comment:
please add a comment about what we are trying to do here, or even an example of how its output is gonna be

Reply:
moved to a non runtime-specific place (builder) with ample comments